src/hooks/wallets/wallets.ts

```diff
@@ -18,7 +17,10 @@ const prefersDarkMode = (): boolean => {
   return window?.matchMedia('(prefers-color-scheme: dark)')?.matches
 }
 
-const walletConnectV2 = (chain: ChainInfo): WalletInit => {
+type WalletInits = InitOptions['wallets']
+type WalletInit = WalletInits extends Array<infer U> ? U : never
```

Comment:
so WalletInit is not an Array of something, but just something?

Reply:
This conditional expression means:
- if `WalletInits` is an array, `WalletInit` is an inferred type of the array's members

So it's a weird way to extract the type of an individual WalletInit from an array.

Reply:
ah, thanks!